### ISSUE TITLE
Event dialog asks for location permission every time it's opened

### ISSUE THREAD

**Issue by mcclurgm** [elementary/calendar#574]
## Prerequisites
- [x] I have searched open and closed issues for duplicates.

## Describe the bug
When location permission is disabled in privacy settings, the event dialog throws up a dialog asking for permission every time it's opened.

### To Reproduce
1. In privacy settings, turn off location services. This can be either completely, or just for Calendar.
2. Open Calendar click the new event button.
3. The event dialog opens, and a permission dialog immediately follows.

You can do these steps any number of times and the permission dialog will appear.

### Expected behavior

Calendar should ask for this permission once, then not send this dialog again. If we want to keep asking for permission, it should be in the UI of the dialog somewhere. Nagging the user every time they edit an event is not a good experience.

### Logs

n/a

## Platform Information

![about](https://user-images.githubusercontent.com/29022469/89348562-622db580-d672-11ea-89a9-7eea1228558c.png)
- [x] I'm using the latest version from git that I've manually compiled
- [ ] I'm using the latest released stable version

## Additional context

n/a

**Comment by fabiozaramella** [comment]
I can't reproduce this issue. Does it still happen to you?

**Comment by mcclurgm** [comment]
Every time (on Hera fwiw). Do you have location permission disabled for calendar?

**Comment by fabiozaramella** [comment]
Yes, it's disabled but I'm on Odin daily so that may be the reason then.

**Comment by 4jNsY6fCVqZv** [comment]
Exciting, first I asked myself how the issue could be reproduced. Tracking services are basically switched off for me.
This leads me to this issue https://github.com/elementary/calendar/issues/540
To be able to report your steps, I first have to activate the tracking services in the Security & Privacy Plug. Then open Calendar and select `Deny`. So I suspect that a system which has deactivated the tracking services from the beginning cannot reproduce the issue.

Another thing I noticed, once there is an entry of Calendar in the Security & Privacy Plug, I can't delete it from the list anymore. I can only deactivate or activate it. Is this intended?
